File: src/main/cpp/commands/command_MoveElevatorManually.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/command_MoveElevatorManually.h"
#include <frc/smartdashboard/SmartDashboard.h>
#include <frc/XboxController.h>

command_MoveElevatorManually::command_MoveElevatorManually(subsystem_Elevator* Elevator, std::function<double()> triggers) :  m_Elevator{Elevator}, m_TriggerInput{triggers} {

  AddRequirements({m_Elevator});
  // Use addRequirements() here to declare subsystem dependencies.

  //Object of subsystem in robotcontainer values assign to use them locally
}

// Called when the command is initially scheduled.
void command_MoveElevatorManually::Initialize() {

  

}

// Called repeatedly when this Command is scheduled to runsdfsdfsdfsdfsdfsdfdf
void command_MoveElevatorManually::Execute() {

  frc::SmartDashboard::SmartDashboard::PutNumber("Manual joystick entered exceute",true);
  frc::SmartDashboard::SmartDashboard::PutNumber("Trigger input val", m_TriggerInput());
  if(fabs(m_TriggerInput() > ControllerConstants::Deadband))
  {
    m_Elevator->MoveElevatorManually(m_TriggerInput());
  }
}

// Called once the command ends or is interrupted.
void command_MoveElevatorManually::End(bool interrupted) {}

// Returns true when the command should end.
bool command_MoveElevatorManually::IsFinished() {
  return false;
}

File: src/main/java/frc/robot/commands/command_MoveElevatorManually.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;

import frc.robot.subsystems.subsystem_Elevator;
import frc.robot.Constants;

import java.util.function.DoubleSupplier;

import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;

public class command_MoveElevatorManually extends CommandBase {
  /** Creates a new command_MoveElevatorManually. */
  private subsystem_Elevator m_Elevator;
  private DoubleSupplier m_TriggerInput;

  public command_MoveElevatorManually(subsystem_Elevator elevator, DoubleSupplier triggerInput) {
    // Use addRequirements() here to declare subsystem dependencies.
    addRequirements(m_Elevator);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {}

  // Called every time the scheduler runs while the command is scheduled.
  @Override
  public void execute() {
    
    SmartDashboard.putBoolean("Manual joystick entered exceute",true);
    SmartDashboard.putNumber("Trigger input val", m_TriggerInput.getAsDouble());

    if(Math.abs(m_TriggerInput.getAsDouble()) > Constants.ControllerConstants.deadband){
      m_Elevator.moveElevatorManually(m_TriggerInput.getAsDouble());
    }
  }

  // Called once the command ends or is interrupted.
  @Override
  public void end(boolean interrupted) {}

  // Returns true when the command should end.
  @Override
  public boolean isFinished() {
    return false;
  }
}
